internal/ingress/controller/controller.go

```diff
@@ -1115,6 +1115,28 @@ func (n *NGINXController) createServers(data []*ingress.Ingress,
 	return servers
 }
 
+func canMergeUpstreams(primary *ingress.Backend, alternative *ingress.Backend) bool {
+	if primary.Name != alternative.Name && !primary.NoServer {
+		return true
+	}
+
+	return false
+}
+
+// Performs the merge action and checks to ensure that one two alternative backends do not merge into each other
+func mergeAlternativeBackend(priUps *ingress.Backend, altUps *ingress.Backend) bool {
```

Comment:
Why are you referring backends as upstreams? When you log below you correctly call them alternative and primary backend.

Reply:
Never mind - looks like they're interchangeably used everywhere in this file. We can revisit this later.

Reply:
however it'd be great if you can make the signature of this function consistent with https://github.com/kubernetes/ingress-nginx/pull/3446/files#diff-86ac9ff9d75a0c5005c116e766a6127dR1118